### PR TITLE
Issue #2657: change function cli getstate to use REST endpoint

### PR DESCRIPTION
**Pull request by jiazhai**
### Motivation

change StateGetter to use REST endpoint

### Modifications

- change StateGetter to use REST endpoint
- change related unit test

### Result

unit tests pass.

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java [hunk 1]
```diff
@@ -661,10 +661,10 @@ void runCmd() throws Exception {
```
Comment (srkukarni) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java]: as part of this pr, can you also get rid of all dependencies from pulsar-admin of any bookkeeper stuff?

Comment (jiazhai) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java]: @srkukarni Thanks. removed un-used imports.

**Comment by jiazhai** [comment]
rerun integration tests

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java (hunk 1 above)
Comment (srkukarni) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java]: @jiazhai I think there are still some references in pom.xml in pulsar-admin about bookkeeper dependencies. Maybe we can now remove them as well?

**Comment by jiazhai** [comment]
rerun java8 tests

**Comment by sijie** [comment]
@srkukarni please review this latest pr.

run java8 tests

**Comment by srkukarni** [comment]
rerun java8 tests

**Comment by jiazhai** [comment]
rebased with latest master

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
@jiazhai  can you fix the license issue?

```
+ src/check-binary-license ./distribution/server/target/apache-pulsar-2.3.0-SNAPSHOT-bin.tar.gz
com.google.code.findbugs-jsr305-3.0.0.jar unaccounted for in LICENSE
commons-lang-commons-lang-2.4.jar unaccounted for in LICENSE
commons-lang-commons-lang-2.6.jar mentioned in LICENSE, but not bundled
com.google.code.findbugs-jsr305-3.0.2.jar mentioned in LICENSE, but not bundled
org.hamcrest-hamcrest-core-1.1.jar mentioned in LICENSE, but not bundled
```

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
rerun cpp tests

**Comment by jiazhai** [comment]
rerun cpp tests

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by jiazhai** [comment]
rerun integration tests

**Pull request merged**